src_py/apiServer/stats.py

```diff
@@ -225,18 +225,36 @@ def build_worker_label_df(original_df, batch_ids, batch_size):
                         confusion_matrix_worker_dict[(worker_name, class_name)] += confusion_matrix
                     
                 else: # Multi-Class
-                    # Take 2 list from the df, one for the actual labels and one for the predict labels to build the confusion matrix
-                    max_column_predict_index = df_worker_labels.iloc[:, num_of_labels:].idxmax(axis=1) 
-                    max_column_predict_index = max_column_predict_index.tolist() 
-                    max_column_predict_index = [int(predict_index) - num_of_labels for predict_index in max_column_predict_index] # fix the index to original labels index
-                    max_column_labels_index = df_worker_labels.iloc[:, :num_of_labels].idxmax(axis=1)
-                    max_column_labels_index = max_column_labels_index.tolist()
+                    #check if there is a sample with more than one predicted label
+                    max_in_row = df_worker_labels.iloc[:, num_of_labels:].max(axis=1)
+                    is_max = df_worker_labels.iloc[:, num_of_labels:].eq(max_in_row, axis=0)  #Get a DataFrame of boolean values where True indicates the maximum value in that row
+                    max_counts = is_max.sum(axis=1)    #Get the number of maximum values in each row
+                    has_multiple_max = max_counts.gt(1).any()  #boolean value: checks if there is at least one row with multiple maximum values in the predict labels
+
+                    if has_multiple_max:
+                        LOG_INFO(f"Worker {worker_name} has at least one sample with multiple predicted labels")
+                        #max_column_predict_index = is_max.apply(lambda row: list(row[row].index), axis=1).tolist()  # Generate a list of lists' each sublist has the index of the maximum value in the row
```

Comment:
Why this line is commented out. Decide to remove or to add it with condition.

Reply:
I removed it.